### ISSUE TITLE
WebSocket?

### ISSUE THREAD

**Issue by vinipsmaker** [TrsstProject/trsst#11]
From the trsst debug output I suppose the application is always asking for the server if there are new items.

Am I right? Do you plan to add support for server-sent events like WebSocket or any other and only using this "pulling" model for compatibility under some circumstances?

**Comment by mmpowers** [comment]
Correct: the basic protocol is intentionally simple/dumb polling so that it's easy to implement new client clients and servers on more platforms with existing frameworks.  Even so, http has a number of features to mitigate the cost of polling, like keep-alive connections and if-none headers, and many http libraries and browsers and servers support these automagically.  Plus there's compatibility with http proxies which gives us access to anonymizers like Tor for free.  

That said, there's plenty of room of innovation, not only in adjusting the frequency of polling and what is polled, but also the use of comet or websockets on a client combined with prospective querying on the server, and I expect those approaches will be required give a more real-time instant notification feel, and it may well be the only way we can scale individual users to subscribing to large numbers (100s) of feeds.

There's a further advantage to dumb polling in that there's no user session tracking so it keeps the server stateless, and this greatly simplifies load-balancing and scalability; just fire up more identical dumb servers.  

Persistent state on the client complicates things, but if the payoff of fewer pings mitigates the increased complexity, then we'll gravitate that way.  But we'll need data first either way.
